rust/src/enip/detect.rs

```diff
@@ -0,0 +1,808 @@
+            }
+            EnipPayload::ListIdentity(lip) if !lip.is_empty() => {
+                if let EnipItemPayload::Identity(li) = &lip[0].payload {
+                    return Some(li.protocol_version);
+                }
+            }
+            _ => {}
+        }
+    }
+    return None;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_protocol_version(
+    tx: *mut std::os::raw::c_void, direction: u8, value: *mut u16,
+) -> bool {
+    let tx = cast_pointer!(tx, EnipTransaction);
+    if let Some(val) = enip_tx_get_protocol_version(tx, direction.into()) {
+        *value = val;
+        return true;
+    }
+    return false;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_capabilities(
+    tx: *mut std::os::raw::c_void, value: *mut u16,
+) -> bool {
+    let tx = cast_pointer!(tx, EnipTransaction);
+    if let Some(ref response) = tx.response {
+        if let EnipPayload::ListServices(lsp) = &response.payload {
+            if !lsp.is_empty() {
+                if let EnipItemPayload::Services(ls) = &lsp[0].payload {
+                    *value = ls.capabilities;
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_revision(
+    tx: *mut std::os::raw::c_void, value: *mut u16,
+) -> bool {
+    let tx = cast_pointer!(tx, EnipTransaction);
+    if let Some(ref response) = tx.response {
+        if let EnipPayload::ListIdentity(lip) = &response.payload {
+            if !lip.is_empty() {
+                if let EnipItemPayload::Identity(li) = &lip[0].payload {
+                    *value = ((li.revision_major as u16) << 8) | (li.revision_minor as u16);
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_identity_status(
+    tx: *mut std::os::raw::c_void, value: *mut u16,
+) -> bool {
+    let tx = cast_pointer!(tx, EnipTransaction);
+    if let Some(ref response) = tx.response {
+        if let EnipPayload::ListIdentity(lip) = &response.payload {
+            if !lip.is_empty() {
+                if let EnipItemPayload::Identity(li) = &lip[0].payload {
+                    *value = li.status;
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_state(tx: *mut std::os::raw::c_void, value: *mut u8) -> bool {
+    let tx = cast_pointer!(tx, EnipTransaction);
+    if let Some(ref response) = tx.response {
+        if let EnipPayload::ListIdentity(lip) = &response.payload {
+            if !lip.is_empty() {
+                if let EnipItemPayload::Identity(li) = &lip[0].payload {
+                    *value = li.state;
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_get_serial(
```

Comment:
I think we should start naming these like C functions..  ie) https://github.com/OISF/suricata/blob/master/rust/src/dns/dns.rs#L883

Reply:
done